### DISCUSSION TITLE
Did anyone got xDebug working with WSL2?

### DISCUSSION THREAD

**Discussion by henryfoster** [php/frankenphp#1037] [Q&A] [answered]
Hi,

frankenphp itself worked out of the box for me using WSL2(ubuntu22.04) but I can't get xdebug working.
It works fine for console commands, but not for http requests. It simply does nothing and does not show any errors.
I also don't see anything in the logs.

I could find these two errors in the https://localhost/_profiler/xdebug

⚠️	[Step Debug] Creating socket for 'host.docker.internal:9003', poll success, but error: Operation now in progress (29).	[⊕](https://xdebug.org/docs/errors#DBG-W-SOCK4)
🛑	[Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port).

But I made sure that host.docker.internal:9003 is reachable from the container, so I don't understand the error.

**Comment by withinboredom** [comment, marked as answer]
Most likely, your debugger isn't set to accept "external connections"; this is the setting in phpstorm:

<img width="738" alt="image" src="https://github.com/user-attachments/assets/451d31c2-b803-4832-8348-38c89fe36a7d">

I suspect vscode has a similar setting somewhere, but I am not familiar with it.

Also, for wsl2/docker, setting remote host to `xdebug://nameserver` or `xdebug://gateway` (the former worked best for me in WSL2's mirrored networking mode).

**Reply by henryfoster** [reply]
Problem was on my end. Somehow I ended up with docker installed via windows (docker desktop) and inside WSL2. Uninstalled both then only installed docker desktop, and it works with PHP storm default settings.
